### ISSUE TITLE
bun install bug

### ISSUE THREAD

**Issue by soneymathew** [oven-sh/bun#4944]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Darwin 22.6.0 arm64 arm

### What steps can reproduce the bug?

```
mkdir bun-npm-alias-test
cd bun-npm-alias-test
bun init
```

use the below package.json
```
{
  "name": "bun-npm-alias-test",
  "module": "index.ts",
  "type": "module",
  "devDependencies": {
    "bun-types": "latest",
    "babel-plugin-relay": "npm:atl-babel-plugin-relay@0.0.0-main-86d355c1",
    "relay-compiler": "npm:atl-relay-compiler@0.0.0-main-86d355c1",
    "relay-test-utils": "npm:atl-relay-test-utils@0.0.0-main-86d355c1"
  },
  "peerDependencies": {
    "typescript": "^5.0.0"
  },
  "resolutions": {
    "relay-compiler": "npm:atl-relay-compiler@0.0.0-main-86d355c1",
    "relay-runtime": "npm:atl-relay-runtime@0.0.0-main-86d355c1",
    "relay-test-utils": "npm:atl-relay-test-utils@0.0.0-main-86d355c1"
  },
  "dependencies": {
    "relay-runtime": "npm:atl-relay-runtime@0.0.0-main-86d355c1"
  }
}
```

### What is the expected behavior?

bun install should succeed

### What do you see instead?

```
$bun install
bun install v1.0.0 (822a00c4)


error: No version matching "0.0.0-main-86d355c1" found for specifier "relay-runtime" (but package exists)
```

### Additional information

Remove 
```
    "relay-compiler": "npm:atl-relay-compiler@0.0.0-main-86d355c1",
    "relay-test-utils": "npm:atl-relay-test-utils@0.0.0-main-86d355c1"
```

from devDependencies

to get `bun install` to succeed.

**Comment by soneymathew** [comment]
output with `1.0.0-debug+682406c42ceee590c2b90900175f0eb1eaed5637` revision on a locally built binary
```
~/learn/bun/packages/debug-bun-darwin-aarch64/bun-debug install --verbose
[SYS] read(3, 4096) = 4096 (0.030ms)
[SYS] close(3)
bun install v1.0.0_debug (682406c4)
Loaded 1 hoisted_dependencies
Loaded 2 resolutions
Loaded 2 dependencies
Loaded 0 extern_strings
Loaded 95 string_bytes

 [0.00ms] [cache hit] atl-babel-plugin-relay
   - "atl-babel-plugin-relay": "npm:atl-babel-plugin-relay@0.0.0-main-86d355c1" - atl-babel-plugin-relay@0.0.0-main-86d355c1

 [0.00ms] [cache hit] atl-relay-compiler
   - "atl-relay-compiler": "npm:atl-relay-compiler@0.0.0-main-86d355c1" - atl-relay-compiler@0.0.0-main-86d355c1

 [0.00ms] [cache hit] atl-relay-test-utils
   - "atl-relay-test-utils": "npm:atl-relay-test-utils@0.0.0-main-86d355c1" - atl-relay-test-utils@0.0.0-main-86d355c1

 [0.00ms] [cache hit] atl-relay-runtime
   - "atl-relay-runtime": "npm:atl-relay-runtime@0.0.0-main-86d355c1" - atl-relay-runtime@0.0.0-main-86d355c1

 [0.00ms] [cache hit] babel-plugin-macros
   - "babel-plugin-macros": "^2.0.0" - babel-plugin-macros@2.8.0

 [0.00ms] [cache hit] cosmiconfig
   - "cosmiconfig": "^5.0.5" - cosmiconfig@5.2.1

 [0.00ms] [cache hit] graphql
   - "graphql": "15.3.0" - graphql@15.3.0

 [0.00ms] [cache hit] @babel/runtime
   - "@babel/runtime": "^7.0.0" - @babel/runtime@7.22.15

 [0.00ms] [cache hit] fbjs
   - "fbjs": "^3.0.2" - fbjs@3.0.5

 [0.00ms] [cache hit] invariant
   - "invariant": "^2.2.4" - invariant@2.2.4

 [0.00ms] [cache hit] relay-runtime
   - "cosmiconfig": "^6.0.0" - cosmiconfig@6.0.0

 [0.00ms] [cache hit] resolve
   - "resolve": "^1.12.0" - resolve@1.22.4

 [0.00ms] [cache hit] import-fresh
   - "import-fresh": "^2.0.0" - import-fresh@2.0.0

 [0.00ms] [cache hit] is-directory
   - "is-directory": "^0.3.1" - is-directory@0.3.1

 [0.00ms] [cache hit] js-yaml
   - "js-yaml": "^3.13.1" - js-yaml@3.14.1

 [0.00ms] [cache hit] parse-json
   - "parse-json": "^4.0.0" - parse-json@4.0.0

 [0.00ms] [cache hit] regenerator-runtime
   - "regenerator-runtime": "^0.14.0" - regenerator-runtime@0.14.0

 [0.00ms] [cache hit] cross-fetch
   - "cross-fetch": "^3.1.5" - cross-fetch@3.1.8

 [0.00ms] [cache hit] fbjs-css-vars
   - "fbjs-css-vars": "^1.0.0" - fbjs-css-vars@1.0.2

 [0.00ms] [cache hit] loose-envify
   - "loose-envify": "^1.0.0" - loose-envify@1.4.0

 [0.00ms] [cache hit] object-assign
   - "object-assign": "^4.1.0" - object-assign@4.1.1

 [0.00ms] [cache hit] promise
   - "promise": "^7.1.1" - promise@7.3.1

 [0.00ms] [cache hit] setimmediate
   - "setimmediate": "^1.0.5" - setimmediate@1.0.5

 [0.00ms] [cache hit] ua-parser-js
   - "ua-parser-js": "^1.0.35" - ua-parser-js@1.0.36

 [0.00ms] [cache hit] @types/parse-json
   - "@types/parse-json": "^4.0.0" - @types/parse-json@4.0.0
   - "import-fresh": "^3.1.0" - import-fresh@3.3.0
   - "parse-json": "^5.0.0" - parse-json@5.2.0

 [0.00ms] [cache hit] path-type
   - "path-type": "^4.0.0" - path-type@4.0.0

 [0.00ms] [cache hit] yaml
   - "yaml": "^1.7.2" - yaml@1.10.2

 [0.00ms] [cache hit] is-core-module
   - "is-core-module": "^2.13.0" - is-core-module@2.13.0

 [0.00ms] [cache hit] path-parse
   - "path-parse": "^1.0.7" - path-parse@1.0.7

 [0.00ms] [cache hit] supports-preserve-symlinks-flag
   - "supports-preserve-symlinks-flag": "^1.0.0" - supports-preserve-symlinks-flag@1.0.0

 [0.00ms] [cache hit] caller-path
   - "caller-path": "^2.0.0" - caller-path@2.0.0

 [0.00ms] [cache hit] resolve-from
   - "resolve-from": "^3.0.0" - resolve-from@3.0.0

 [0.00ms] [cache hit] argparse
   - "argparse": "^1.0.7" - argparse@1.0.10

 [0.00ms] [cache hit] esprima
   - "esprima": "^4.0.0" - esprima@4.0.1

 [0.00ms] [cache hit] error-ex
   - "error-ex": "^1.3.1" - error-ex@1.3.2

 [0.00ms] [cache hit] json-parse-better-errors
   - "json-parse-better-errors": "^1.0.1" - json-parse-better-errors@1.0.2

 [0.00ms] [cache hit] node-fetch
   - "node-fetch": "^2.6.12" - node-fetch@2.7.0

 [0.00ms] [cache hit] js-tokens
   - "js-tokens": "^3.0.0 || ^4.0.0" - js-tokens@4.0.0

 [0.00ms] [cache hit] asap
   - "asap": "~2.0.3" - asap@2.0.6

 [0.00ms] [cache hit] parent-module
   - "parent-module": "^1.0.0" - parent-module@1.0.1
   - "resolve-from": "^4.0.0" - resolve-from@4.0.0

 [0.00ms] [cache hit] @babel/code-frame
   - "@babel/code-frame": "^7.0.0" - @babel/code-frame@7.22.13

 [0.00ms] [cache hit] json-parse-even-better-errors
   - "json-parse-even-better-errors": "^2.3.0" - json-parse-even-better-errors@2.3.1

 [0.00ms] [cache hit] lines-and-columns
   - "lines-and-columns": "^1.1.6" - lines-and-columns@1.2.4

 [0.00ms] [cache hit] has
   - "has": "^1.0.3" - has@1.0.3

 [0.00ms] [cache hit] caller-callsite
   - "caller-callsite": "^2.0.0" - caller-callsite@2.0.0

 [0.00ms] [cache hit] sprintf-js
   - "sprintf-js": "~1.0.2" - sprintf-js@1.0.3

 [0.00ms] [cache hit] is-arrayish
   - "is-arrayish": "^0.2.1" - is-arrayish@0.2.1

 [0.00ms] [cache hit] whatwg-url
   - "whatwg-url": "^5.0.0" - whatwg-url@5.0.0

 [0.00ms] [cache hit] callsites
   - "callsites": "^3.0.0" - callsites@3.1.0

 [0.00ms] [cache hit] @babel/highlight
   - "@babel/highlight": "^7.22.13" - @babel/highlight@7.22.13

 [0.00ms] [cache hit] chalk
   - "chalk": "^2.4.2" - chalk@2.4.2

 [0.00ms] [cache hit] function-bind
   - "function-bind": "^1.1.1" - function-bind@1.1.1
   - "callsites": "^2.0.0" - callsites@2.0.0

 [0.00ms] [cache hit] tr46
   - "tr46": "~0.0.3" - tr46@0.0.3

 [0.00ms] [cache hit] webidl-conversions
   - "webidl-conversions": "^3.0.0" - webidl-conversions@3.0.1

 [0.00ms] [cache hit] @babel/helper-validator-identifier
   - "@babel/helper-validator-identifier": "^7.22.5" - @babel/helper-validator-identifier@7.22.15

 [0.00ms] [cache hit] ansi-styles
   - "ansi-styles": "^3.2.1" - ansi-styles@3.2.1

 [0.00ms] [cache hit] escape-string-regexp
   - "escape-string-regexp": "^1.0.5" - escape-string-regexp@1.0.5

 [0.00ms] [cache hit] supports-color
   - "supports-color": "^5.3.0" - supports-color@5.5.0

 [0.00ms] [cache hit] color-convert
   - "color-convert": "^1.9.0" - color-convert@1.9.3

 [0.00ms] [cache hit] has-flag
   - "has-flag": "^3.0.0" - has-flag@3.0.0

 [0.00ms] [cache hit] color-name
   - "color-name": "1.1.3" - color-name@1.1.3


error: No version matching "0.0.0-main-86d355c1" found for specifier "relay-runtime" (but package exists)
```

**Comment by soneymathew** [comment]
TIL React experimental builds use a similar versioning strategy too
https://www.npmjs.com/package/react/v/0.0.0-experimental-bbc8530ed-20230912?activeTab=versions

**Comment by soneymathew** [comment]
Double checked if #6312 fixes this with revision 1.0.5-debug+25cf4016d34af21c5fdd76d0f7d58958ce19d882.

**Comment by dylan-conway** [comment]
fixed by #6435 and #6545

**Comment by soneymathew** [comment]
@dylan-conway  Thanks for fixing this. 
Tested that the fix works, feel free to close this issue.
